src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceLocator.cs

```diff
@@ -108,6 +108,22 @@ public static void InitializePackageServiceProvider(IServiceProvider provider)
             return Package.GetGlobalService(typeof(TService)) as TInterface;
         }
 
+        public static TInterface GetGlobalServiceFreeThreaded<TService, TInterface>() where TInterface : class
```

Comment:
Do you need this new one? :) 
I have a PR to replace IServiceProvider with IAsyncServiceProvider https://github.com/NuGet/NuGet.Client/pull/2898. 

That would mean that this call cannot be synchrnous.

If possible, please avoid this.

Reply:
I see from the related code that it would be challenging to do, as there are both synchronous and asynchronous invocations.

Reply:
Yes, we need this free-threaded version.  It is unnecessary to switch to the main thread to get the `IVsInvalidateCachedCommandState` service, and we should avoid unnecessary switching.

We can discuss async or not offline.